Get framebuffer resolution on more platforms

media_driver/linux/common/codec/ddi/media_libva_decoder.cpp:48:10: fatal error: 'linux/fb.h' file not found
#include <linux/fb.h>
         ^~~~~~~~~~~~
media_driver/linux/common/codec/ddi/media_libva_decoder.cpp:55:43: error: variable has incomplete type 'struct fb_var_screeninfo'
    struct fb_var_screeninfo              vsinfo;
                                          ^
media_driver/linux/common/codec/ddi/media_libva_decoder.cpp:62:22: error: use of undeclared identifier 'FBIOGET_VSCREENINFO'
        if(ioctl(fd, FBIOGET_VSCREENINFO, &vsinfo) < 0)
                     ^
media_driver/linux/common/ddi/media_libva.cpp:39:10: fatal error: 'linux/fb.h' file not found
#include <linux/fb.h>
         ^~~~~~~~~~~~

--- media_driver/linux/common/codec/ddi/media_libva_decoder.cpp.orig	2018-10-01 21:39:52 UTC
+++ media_driver/linux/common/codec/ddi/media_libva_decoder.cpp
@@ -45,7 +45,39 @@
 #include <X11/Xutil.h>
 #endif
 
+#if defined(__linux__)
 #include <linux/fb.h>
+#define DEFAULT_FBDEV "/dev/graphics/fb0"
+#elif defined(__DragonFly__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__sun)
+#include <sys/fbio.h>
+# if defined(__sun)
+#define DEFAULT_FBDEV "/dev/fb"
+# else
+#define DEFAULT_FBDEV "/dev/ttyv0"
+# endif
+#define FBIOGET_VSCREENINFO FBIOGTYPE
+#define fb_var_screeninfo fbtype
+#define xres fb_width
+#define yres fb_height
+#elif defined(__NetBSD__) || defined(__OpenBSD__)
+#include <dev/wscons/wsconsio.h>
+# if defined(__OpenBSD__)
+#define DEFAULT_FBDEV "/dev/ttyC0"
+# else
+#define DEFAULT_FBDEV "/dev/ttyE0"
+# endif
+#define FBIOGET_VSCREENINFO WSDISPLAYIO_GINFO
+#define fb_var_screeninfo wsdisplay_fbinfo
+#define xres width
+#define yres height
+#else
+#define DEFAULT_FBDEV "/dev/fb0"
+#define FBIOGET_VSCREENINFO 0
+struct fb_var_screeninfo {
+    uint32_t xres;
+    uint32_t yres;
+};
+#endif
 
 typedef MediaDdiFactory<DdiMediaDecode, DDI_DECODE_CONFIG_ATTR> DdiDecodeFactory;
 static int32_t DdiDecode_GetDisplayInfo(VADriverContextP ctx)
@@ -56,7 +88,7 @@ static int32_t DdiDecode_GetDisplayInfo(VADriverContex
     vsinfo.xres                           = 0;
     vsinfo.yres                           = 0;
 
-    fd = open("/dev/graphics/fb0",O_RDONLY);
+    fd = open(DEFAULT_FBDEV,O_RDONLY);
     if(fd > 0)
     {
         if(ioctl(fd, FBIOGET_VSCREENINFO, &vsinfo) < 0)
--- media_driver/linux/common/ddi/media_libva.cpp.orig	2018-11-26 07:05:55 UTC
+++ media_driver/linux/common/ddi/media_libva.cpp
@@ -36,8 +36,6 @@
 #include <X11/Xutil.h>
 #endif
 
-#include <linux/fb.h>
-
 #include "media_libva.h"
 
 #include "media_libva_util.h"
